src/main/java/com/minecolonies/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -0,0 +1,307 @@
+            return AIState.GUARD_HUNT_DOWN_TARGET;
+        }
+    }
+
+    /**
+     * Searches for the next taget.
+     * @return the next AIState.
+     */
+    protected AIState searchTarget()
+    {
+        entityList = this.worker.worldObj.getEntitiesWithinAABB(EntityMob.class, this.getTargetableArea(currentSearchDistance));
+        entityList.addAll(this.worker.worldObj.getEntitiesWithinAABB(EntitySlime.class, this.getTargetableArea(currentSearchDistance)));
+        entityList.addAll(this.worker.worldObj.getEntitiesWithinAABB(EntityPlayer.class, this.getTargetableArea(currentSearchDistance)));
+
+        setDelay(BASE_DELAY);
+        if(entityList.isEmpty())
+        {
+            if(currentSearchDistance < getMaxVision())
+            {
+                currentSearchDistance += START_SEARCH_DISTANCE;
+            }
+            else
+            {
+                currentSearchDistance = START_SEARCH_DISTANCE;
+                return AIState.GUARD_PATROL;
+            }
+
+            return AIState.GUARD_SEARCH_TARGET;
+        }
+        return AIState.GUARD_GET_TARGET;
+    }
+
+    /**
+     * Getter for the vision or attack distance.
+     * @return the max vision.
+     */
+    private double getMaxVision()
+    {
+        BuildingGuardTower guardTower = (BuildingGuardTower) worker.getWorkBuilding();
+        return (guardTower == null) ? 0 : (MAX_ATTACK_DISTANCE + guardTower.getBonusVision());
+    }
+
+    /**
+     * Getter calculating how many arrows the guard may shoot or deal sword hits until restock.
+     * @return the amount.
+     */
+    protected int getMaxAttacksUntilRestock()
+    {
+        BuildingGuardTower guardTower = (BuildingGuardTower) worker.getWorkBuilding();
+        return (guardTower == null) ? 0 : (MAX_ATTACKS + guardTower.getBuildingLevel());
+    }
+
+    /**
+     * Lets the guard patrol inside the colony area searching for mobs.
+     * @return the next state to go.
+     */
+    protected AIState patrol()
+    {
+        worker.setAIMoveSpeed(1);
+
+        if(currentPatrolTarget == null)
+        {
+            currentPatrolTarget = getRandomBuilding();
+        }
+
+        if(worker.isWorkerAtSiteWithMove(currentPatrolTarget, PATH_CLOSE))
+        {
+            currentPatrolTarget = null;
+        }
+
+        return AIState.GUARD_SEARCH_TARGET;
+    }
+
+    /**
+     * Gets a random building from his colony.
+     * @return a random blockPos.
+     */
+    private BlockPos getRandomBuilding()
+    {
+        if(worker.getColony() == null || getOwnBuilding() == null)
+        {
+            return worker.getPosition();
+        }
+
+        Map<BlockPos, AbstractBuilding> buildingMap = worker.getColony().getBuildings();
+        int random = worker.getRandom().nextInt(buildingMap.size());
+
+        AbstractBuilding building = (AbstractBuilding) worker.getColony().getBuildings().values().toArray()[random];
+        if(building instanceof BuildingGuardTower)
```

Comment:
Why not patrol to other Guard Towers? Have guards check on each other? That would bring live imo